tests/unit/test_finder.py

```diff
@@ -34,6 +34,15 @@ def test_no_partial_name_match(data):
     assert found.url.endswith("gmpy-1.15.tar.gz"), found
 
 
+@patch('pip.index.os.path.exists')
+def test_tilde(data):
```

Comment:
While I think from reading the mock documentation that patch defaults to setting the patched object to something that returns true when called, it would be a lot clearer if you made it explicit that you're patching ```os.path.exists``` to be ```lambda: True```.